Regression in 2.3.3.251201

**Describe the bug**
The following code snippet runs and passes `mypy` type checking w/ `pandas-stubs==2.3.2.250926` (second-latest release at time of writing). However, it fails w/ `pandas-stubs==2.3.3.251201` (latest release at time of writing).

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.

Code:
```py
from datetime import UTC, datetime, timedelta
import pandas as pd

dt1 = datetime.now(UTC)
dt2 = dt1 + timedelta(seconds=5)
df = pd.DataFrame((dt1, dt2), columns=("time",))
diff = df["time"].diff()
dt = diff.dt  # First error here
print(dt.total_seconds())  # Second error here
```

Output when run:
```
0    NaN
1    5.0
Name: time, dtype: float64
```

2. Indicate which type checker you are using (`mypy` or  `pyright`).

`mypy`

3. Show the error message received from that type checker while checking your example.

```
pd.py:8: error: Argument 1 to "__get__" of "DtDescriptor" has incompatible type "Series[float]"; expected "Series[Never]"  [arg-type]
pd.py:9: error: "Properties" has no attribute "total_seconds"  [attr-defined]
Found 2 errors in 1 file (checked 1 source file)
```

**Please complete the following information:**
 - OS: Linux
 - OS Version: Arch Linux x86_64
 - python version: 3.13.7
 - version of type checker: mypy 1.19.0 (compiled: yes)
 - version of installed `pandas-stubs`: 2.3.3.251201


**Additional context**
```
$ pip list
Package           Version
----------------- -----------------
librt             0.6.3
mypy              1.19.0
mypy_extensions   1.1.0
numpy             2.3.5
pandas            2.3.3
pandas-stubs      2.3.3.251201
pathspec          0.12.1
pip               25.3
python-dateutil   2.9.0.post0
pytz              2025.2
six               1.17.0
types-pytz        2025.2.0.20251108
typing_extensions 4.15.0
tzdata            2025.2
```

--- a/pandas-stubs/core/indexes/accessors.pyi
+++ b/pandas-stubs/core/indexes/accessors.pyi
@@ -446,7 +446,7 @@ class DtDescriptor:
     @overload
     def __get__(
         self, instance: Series[Never], owner: type[Series]
-    ) -> TimestampProperties | TimedeltaProperties | PeriodProperties: ...
+    ) -> CombinedDatetimelikeProperties: ...
     @overload
     def __get__(
         self, instance: Series[Timestamp], owner: type[Series]

--- a/tests/series/test_properties.py
+++ b/tests/series/test_properties.py
@@ -8,6 +8,7 @@ from pandas.core.arrays.interval import IntervalArray
 from pandas.core.arrays.timedeltas import TimedeltaArray
 from pandas.core.frame import DataFrame
 from pandas.core.indexes.accessors import (
+    CombinedDatetimelikeProperties,
     DatetimeProperties,
     PeriodProperties,
     TimedeltaProperties,
@@ -52,15 +53,28 @@ def test_property_dt() -> None:
     # python/mypy#19952: mypy gives Any
     check(
         assert_type(  # type: ignore[assert-type]
-            df["ts"].dt, "TimestampProperties | TimedeltaProperties | PeriodProperties"
+            df["ts"].dt, CombinedDatetimelikeProperties
         ),
         DatetimeProperties,
     )
     check(
         assert_type(  # type: ignore[assert-type]
-            df["td"].dt, "TimestampProperties | TimedeltaProperties | PeriodProperties"
+            df["td"].dt, CombinedDatetimelikeProperties
         ),
         TimedeltaProperties,
+    )
+
+    check(
+        assert_type(df["ts"].dt.year, "Series[int]"),  # type: ignore[assert-type]
+        Series,
+        np.integer,
+    )
+    check(
+        assert_type(  # type: ignore[assert-type]
+            df["td"].dt.total_seconds(), "Series[float]"
+        ),
+        Series,
+        np.floating,
     )
 
     if TYPE_CHECKING_INVALID_USAGE:

